ethcore/src/client/client.rs

```diff
@@ -412,8 +410,10 @@ impl Importer {
 
 		let is_epoch_begin = chain.epoch_transition(parent.number(), *header.parent_hash()).is_some();
 
-		let enact_result = enact_verified(
-			block,
+		let enact_result = enact(
```

Comment:
Calling `enact()` directly here lets me avoid splitting up the `PreverifiedBlock` before making the call (and pass the `Header` by ref).

Reply:
very neat

Reply:
I guess `enact_verified` added a bit of type safety to `enact`. Another approach would be to keep that function, but change the return type to returned unused but consumed fields of `PreverifiedBlock`, something like `EthcoreResult<(LockedBlock, Option<_>, Bytes)>`.

Reply:
Yeah I tried that, and Niklas did too I think. It gets pretty nasty and I'd rather not. It feels like a dirty trick to hand in data and ask for it back.

Reply:
Yeah, it gets messy because we then we need to return `Bytes` in both the `Ok` and `Err` case because of this damn `BadBlocks::report`

Reply:
Ok, we can merge the PR as is, I can take a look if it could be cleaned up.